exec/cnex/cnex.c

```diff
@@ -237,6 +248,9 @@ void exec_raiseLiteral(TExecutor *self, TString *name, TString *info, Number cod
                 TString *handler = self->object->strings[self->object->exceptions[i].exid];
                 if ((string_compareString(name, handler) == 0) || (name->length > handler->length && string_startsWith(name, handler) && name->data[handler->length] == '.')) {
                     self->ip = self->object->exceptions[i].handler;
+                    while (self->stack->top > ((framestack_isEmpty(self->framestack) ? 0 : self->framestack->top) + (int32_t)self->object->exceptions[i].stack_depth)) {
```

Comment:
This doesn't look right. What is `self->framestack->top`? I thought the member you added was called `stack_depth`. Consider changing the name of `top` to reflect what it really is the top of (I don't know what that is).

Reply:
Removed ambiguity with stack_depth member.  `top` points to the top most item on the stack.

Reply:
This still isn't right. You're using the depth of the framestack (`self->framestack->top`) instead of the `stack_depth` of the frame on top of the framestack.

Reply:
Fixed this to look at the proper stack frame.